contracts/crosschain/ERC7786Aggregator.sol

```diff
@@ -0,0 +1,281 @@
+            if (id != bytes32(0)) {
+                outbox[i] = Outbox(gateway, id);
+                needsId = true;
+            }
+        }
+
+        if (needsId) {
+            outboxId = keccak256(abi.encode(outbox));
+            emit OutboxDetails(outboxId, outbox);
+        }
+
+        emit MessagePosted(
+            outboxId,
+            CAIP10.local(msg.sender),
+            CAIP10.format(destinationChain, receiver),
+            payload,
+            attributes
+        );
+    }
+
+    // ============================================== IERC7786Receiver ===============================================
+
+    /// @inheritdoc IERC7786Receiver
+    function executeMessage(
+        string calldata sourceChain, // CAIP-2 chain identifier
+        string calldata sender, // CAIP-10 account address (does not include the chain identifier)
+        bytes calldata payload,
+        bytes[] calldata attributes
+    ) public payable virtual whenNotPaused returns (bytes4) {
+        // Check sender is a trusted remote aggregator
+        if (!_remotes[sourceChain].equal(sender)) revert ERC7786AggregatorInvalidCrosschainSender();
+
+        // Message reception tracker
+        bytes32 id = keccak256(abi.encode(sourceChain, sender, payload, attributes));
+        Tracker storage tracker = _trackers[id];
+
+        // If call originates from a trusted gateway
+        if (_gateways.contains(msg.sender) && !tracker.receivedBy[msg.sender]) {
+            // Count number of time received
+            tracker.receivedBy[msg.sender] = true;
+            ++tracker.countReceived;
+            emit Received(id, msg.sender);
+
+            // if already executed, leave gracefully
+            if (tracker.executed) return IERC7786Receiver.executeMessage.selector;
+        } else {
+            // if already executed, revert
+            if (tracker.executed) revert ERC7786AggregatorAlreadyExecuted();
+        }
+
+        // Parse payload
+        (, string memory originalSender, string memory receiver, bytes memory unwrappedPayload) = abi.decode(
+            payload,
+            (uint256, string, string, bytes)
+        );
+
+        // If ready to execute, and not yet executed
+        if (tracker.countReceived >= getThreshold()) {
+            // prevent re-entry
+            tracker.executed = true;
+
+            bytes memory call = abi.encodeCall(
+                IERC7786Receiver.executeMessage,
+                (sourceChain, originalSender, unwrappedPayload, attributes)
+            );
+            // slither-disable-next-line reentrancy-no-eth
+            (bool success, bytes memory returndata) = receiver.parseAddress().call(call);
+
+            if (!success) {
+                // rollback to enable retry
+                tracker.executed = false;
+                emit ExecutionFailed(id);
+            } else if (
+                returndata.length >= 32 && abi.decode(returndata, (bytes4)) == IERC7786Receiver.executeMessage.selector
```

Comment:
```suggestion
                abi.decode(returndata, (bytes4)) == IERC7786Receiver.executeMessage.selector
```

I understand that ABI decoding could revert, but length < 32 is not the only reason, it could also have nonzero padding. I don't think we should check for every condition so we should just let it fail.